### ISSUE TITLE
Documentation on buiding docs is outdated

### ISSUE THREAD

**Issue by AlexHls** [tardis-sn/tardis#2089]
**Describe the bug**

The documentation https://tardis-sn.github.io/tardis/contributing/development/documentation_guidelines.html#sharing-the-built-documentation-in-your-pr-documentation-preview says a commit with
`git commit --allow-empty -m "[build docs]"`
will build the docs. The workflow file requires `build-docs` in the commit message.
Furthermore, the documentation does not build either way

**To Reproduce**

See PR https://github.com/tardis-sn/tardis/pull/2088

**Screenshots**

**System**
 - OS:
    - [x] GNU/Linux
    - [x] macOS

 - Environment (`conda list`):

**Additional context**

**Comment by epassaro** [comment]
I would remove the documentation build section since the pull request template says clearly what you need to do in order to build the docs (just apply the `build-docs` label).

**Comment by smithis7** [comment]
> 

I strongly disagree since we still need to document where to access the built documentation, how to view build logs, and how to build docs online if the branch isn't part of an open PR.
